src/main/java/org/reactome/release/qa/diagram/DiagramDrugColorCheck.java

```diff
@@ -0,0 +1,112 @@
+
+    public DiagramDrugColorCheck() {
+    }
+
+    @Override
+    public QAReport executeQACheck() throws Exception {
+        QAReport report = new QAReport();
+        Collection<GKInstance> pathwayDiagrams = getPathwayDiagrams();
+        DiagramGKBReader reader = new DiagramGKBReader();
+        for (GKInstance diagram : pathwayDiagrams) {
+            checkPathwayDiagram(diagram, reader, report);
+        }
+        report.setColumnHeaders("PathwayDiagram_DBID",
+                                "Pathway_DisplayName",
+                                "Pathway_DBID",
+                                "Renderable_DisplayName",
+                                "Renderable_DBID",
+                                "MostRecentAuthor");
+        return report;
+    }
+
+    private void checkPathwayDiagram(GKInstance diagram, DiagramGKBReader reader, QAReport report) throws Exception {
+        RenderablePathway pathway = reader.openDiagram(diagram);
+        String modDate = QACheckerHelper.getLastModificationAuthor(diagram);
+        List<Renderable> components = pathway.getComponents();
+        if (components == null || components.size() == 0)
+            return;
+        // For all rendered components.
+        for (Renderable component : components) {
+            // Check correctness of drug background.
+            List<Class<? extends Node>> classesToCheck = Arrays.asList(RenderableEntitySet.class,
+                                                                       RenderableComplex.class,
+                                                                       RenderableChemicalDrug.class,
+                                                                       RenderableProteinDrug.class,
+                                                                       RenderableRNADrug.class);
+            Class<Renderable> cls = (Class<Renderable>) component.getClass();
+            if (classesToCheck.contains(cls) && !hasCorrectColors(component)) {
+                // Add line to report.
+                report.addLine(diagram.getDBID().toString(),
+                               pathway.getDisplayName(),
+                               pathway.getReactomeDiagramId().toString(),
+                               component.getDisplayName(),
+                               component.getReactomeId().toString(),
+                               modDate);
+            }
+        }
+    }
+
+    /**
+     * Check if a component has the appropriate background color for its drug status.
+     *
+     * Returns true is color matches the status.
+     * Returns false if there is a mismatch error.
+     *
+     * @param component
+     * @return boolean
+     * @throws Exception
+     */
+    private boolean hasCorrectColors(Renderable component) throws Exception {
+        GKInstance instance = dba.fetchInstance(component.getReactomeId());
+        if (component.getReactomeId() == null || instance == null)
```

Comment:
if component.getReactomeId() is null, the statement above this will throw an exception. These two statements logically are not compatible.